### ISSUE TITLE
Add Resistance & Immunity Option

### ISSUE THREAD

**Issue by kajishun** [illandril/FoundryVTT-token-tooltips#81]
**Is your feature request related to a problem? Please describe.**
I am frustrated when trying to add resistances and immunities to my token tooltips. Adding them makes my tooltips ginormous and unwieldy. Having the resistance and immunity data information readily available is very useful as a DM.

**Describe the solution you'd like**
I would love an option to change the wording to simple icons representing the damage types (e.g., lightning bolt for lightning, small flame for fire, water drop for water, etc. Perhaps using the DDB standard icons for clarity is best. Maybe also color them?
 
**Describe alternatives you've considered**
I have tried alternate mods but yours is the best and I'd like to continue using this one.

**What system(s) do you want this feature to work with?**
D&D 5E.

**What other modules do you use that you want to ensure work well with this feature?**
I don't think there are any integrations or deconfliction needed?

**Additional context**
None that I can think of.

**Comment by illandril** [comment]
The DDB icons are copyrighted, so I can't use those.


Here are the mappings I've come up with so far:
![Screenshot (8)](https://github.com/illandril/FoundryVTT-token-tooltips/assets/469205/9364ded1-4fcb-4663-95ab-4f1a7300ac20)

With the names turned off (for the types that have an icon):
![Screenshot (9)](https://github.com/illandril/FoundryVTT-token-tooltips/assets/469205/ffd958b4-0306-44d2-8773-267480e96114)

And here are Pathfinder 2e's versions:
![Screenshot (7)](https://github.com/illandril/FoundryVTT-token-tooltips/assets/469205/1fcdfa1f-9ba0-4815-8a54-e9ac490b73c5)


Since you're the one who asked for the feature, I'll give you the chance now to suggest other icons and/or colors to use.

Foundry ships with [FontAwesome icons](https://fontawesome.com/search), so I'd prefer to use those, but I'm open to including icons that have open-source friendly licensing for any of the damage types or status effects that don't have a nice FontAwesome icon to use (but you'd have to find them - I don't have the time to go searching through icon libraries).

**Comment by kajishun** [comment]
Joe,

Howdy! This is fantastic; thank you for picking up my feature request! I've
Cc'ed Emily Eakes, a professional graphic artist and my girlfriend and a
D&D player/DM, to help us on this. We love the idea of FontAwesome icons-
Is Foundry limited to the free FA icons?

Cheers,
Craig

**Comment by kajishun** [comment]
Hi there,
I'm attaching a PDF with our suggested Font Awesome icons, most are in the
free set, but 3 or 4 were premium. If that's a problem, let us know. I have
included the icon in the suggested color along with a screenshot from Font
Awesome with their name. Below that you'll see the screenshot you shared on
the left. On the right are all the labels with the hex codes for the color
suggestions - several are tweaks on what you already had with the idea of
trying to distinguish the colors a bit more while still allowing for enough
contrast against the gray background.

Bludgeoning, Piercing, and Slashing are the same gray because the icons are
distinct enough, and at a glance, that gray will represent weapon damage.

Blinded and Deafened are the same gray because they're similar types of
conditions. Exhausted and Petrified are the same gray because gray makes
sense for them as well and keeping it simple with one gray made sense.

Poison damage and Poisoned condition have the same icon because it's
universal and easier to read at a tiny size than the bottle with the skull.
And because they will be located under Resistances or Cond. Imm. it should
be clear its poison damage vs. the poisoned condition.

All icons appear to have a consistent height, we're suggesting prone be the
person on its side — looks like that's possible, but if not we can look for
something else.

Let me know if you have any questions. Thank you for reaching out!
emily

**Comment by illandril** [comment]
The PDF didn't attach. I don't think GitHub supports adding attachments to issues via email - you may need to add it through the Web UI.

> most are in the free set, but 3 or 4 were premium. If that's a problem, let us know.
It looks like Foundry bundles the FontAwesome 6 Pro, Brands, Duotone, and Sharp icons, so the premium icons should work.

> Poison damage and Poisoned condition have the same icon because it's universal and easier to read at a tiny size than the bottle with the skull. And because they will be located under Resistances or Cond. Imm. it should be clear its poison damage vs. the poisoned condition.
While true for D&D 5e, that isn't true for other systems, and I want the icons to work for all systems. Pathfinder 2e, for example, bundles all immunities together.

> All icons appear to have a consistent height, we're suggesting prone be the person on its side — looks like that's possible, but if not we can look for something else.
I've got full flexibility on what shows in the tooltip, including any CSS rules that might apply to any of the icons, so rotating (or scaling, or overlaying one on top of another, or two side-by-side, etc) is definitely an option.

**Comment by kajishun** [comment]
+ Emily

**Comment by filthedesign** [comment]
> It looks like Foundry bundles the FontAwesome 6 Pro, Brands, Duotone, and Sharp icons, so the premium icons should work.

Excellent!
> 
> While true for D&D 5e, that isn't true for other systems, and I want the icons to work for all systems. Pathfinder 2e, for example, bundles all immunities together.

If they need to be different that's cool, but looking at the Pathfinder 2e screenshot the skull and crossbones for both immunities and resistances and I'm not seeing a distinction between damage and the condition? But I'm not a Pathfinder player so do whatever makes the most sense there. 
> 
> I've got full flexibility on what shows in the tooltip, including any CSS rules that might apply to any of the icons, so rotating (or scaling, or overlaying one on top of another, or two side-by-side, etc) is definitely an option.

Sounds good, the other part of that you will hopefully be able to see in the attachment this time :) i sized everything at 30px for height, but could totally scale prone down and petrified up for a more consistent feel.

Thanks!
emily

[foundry-markers.pdf](https://github.com/illandril/FoundryVTT-token-tooltips/files/11992482/foundry-markers.pdf)

**Comment by illandril** [comment]
Some of the suggested colors didn't have enough contrast, so I deviated from your suggested colors a little to ensure the contrast ratio against the darkest possible background was at least 3:1 (the tooltip background is a semi-transparent white, so if the tooltip appears over black it is effectively `#ccc`).

I kept `face-vomit` for poisoned, but switched poison to `skull-crossbones` to ensure systems that can have both poison and poisoned in the same list can still distinguish between the two if they use just icons.

Other than those minor tweaks, I liked your suggestions and followed them.
